.github/workflows/acceptance_tests_kind.yaml

```diff
@@ -5,7 +5,7 @@ on:
     inputs:
       kindVersion:
         description: The kind version
-        default: 0.20.0 # Kubernetes version: 1.27.X
+        default: 0.20.0
```

Comment:
Really great stuff in your comments! We would love to take in your fixes involving the test failures in old k8s versions.

Only question I have in this PR is leaving it at `0.20`. Was their a specific reason for not moving it up to the latest in order to test with `1.30` as well?

We are planning to include `1.30` in order to include some new fields in a few resources. Would upgrading the kind version result in more failing tests? Not sure if you tried this already or not.

Reply:
I'll go ahead and get the issues written up and start on PRs for the fixes 👍 

On updating kind, I ran against kind 0.23 locally just fine so I don't expect any real issues, I was mostly trying to keep the scope of this change down since kind is used in a couple other workflows as well and I assume we'd want to keep those in sync. It requires bumping the version of kind **and** updating the Kubernetes versions since the node images are specific to a version of kind. I'm happy to tackle that as another PR or as part of this, it definitely makes sense to me to upgrade.

Reply:
Another note would be that kind v0.23.0 does not support as old of versions of k8s as the test currently run against. It has supported node images going back to 1.25, but not 1.23.  Kubernetes 1.23 is almost 3 years old though so maybe dropping it from the tests makes sense.

Reply:
Yep let's treat that as another PR. We can perform the kind bump once the PR of bumping kubernetes to `v1.30.3` gets merged: https://github.com/hashicorp/terraform-provider-kubernetes/pull/2561

This will also give us a chance to address the issues you discovered prior to bumping to `v1.30.3` 😄